src/transformers/modeling_utils.py

```diff
@@ -5083,6 +5083,24 @@ def loss_function(self):
             loss_type = "ForCausalLM"
         return LOSS_MAPPING[loss_type]
 
+    @property
+    def compiled_call(self):
+        """Return a `torch.compile`'d version of `self.__call__`. This is useful to dynamically choose between
+        non-compiled/compiled `forward` during inference, especially to switch between prefill (where we don't
+        want to use compiled version to avoid recomputing the graph with new shapes) and iterative decoding
+        (where we want the speed-ups of compiled version with static shapes)."""
+        if not hasattr(self, "_compiled_call"):
+            self._last_compile_config = getattr(self.generation_config, "compile_config", CompileConfig())
+            self._compiled_call = torch.compile(self.__call__, **self._last_compile_config.to_dict())
+        return self._compiled_call
+
+    def _set_compile_call(self, compile_config: CompileConfig):
+        """Set the mode (arguments) for the compiled version of `self.__call__`."""
+        # Only reset it if different from previous config
+        if getattr(self, "_last_compile_config", CompileConfig()) != compile_config:
+            self._last_compile_config = compile_config
+            self._compiled_call = torch.compile(self.__call__, **compile_config.to_dict())
```

Comment:
I am wondering if it would (probably) make more sense to do

> self._compiled_call = None

and delegate the actually compile only in `compiled_call` (as you already there) - when it is called (like you do in `generate`).

Having 2 places to call `torch.compile`  seems a slight strange to me (but it's ok though).

And if you decide to take my suggestion, probably the name `_set_compile_call` should be changed to `_set_compile_config`

Reply:
torch.compile does not compile, calling the compile funciton does compile

Reply:
I know. But what I mean here is not the underlying compile, but rather the call to `torch.compile` twice. 2 different methods calling `torch.compile` seems to me the not very logic (for me, it's better if one method is only set config and another one take the job to call `torch.compile`)

But it's no big deal but my habit of making each method does its own job.

Reply:
You're right, but maybe then using `_set_compile_call` in `compiled_call` instead? That way, `set_compile_call` is the only place using `torch.compile`, and `compiled_call` is still only used for accessing the fnction?

Reply:
even with that, in `generate` we still need to call `_set_compile_call` then `compiled_call` right? (as `compiled_call`) doesn't contain the argument. If this is the case, then it's odd to have `compiled_call` calling `_set_compile_call`.

Otherwise, if you think it's ok/good to change `compiled_call` to accept compile_config arugment, then sound good to me with your suggested change.

Reply:
oh, `compiled_call` is a property so not to take argument ....

Reply:
anyway, it's implementation details and not affecting users. Don't take it too serious if the changes will take too much time.

Reply:
Yes, I thought it makes sense to make it a property so that `model.compiled_call(**inputs)` could always be used directly as an alternative `forward(**inputs)`